manifests/ca.pp

```diff
@@ -1,21 +1,56 @@
+# @summary
+#  Creates per CA configuration files.
+#
+# @example
+#  fetchcrl::ca{'EDG-Tutorial-CA':
+#    agingtolerance => 168
+#  }
+#
+# @param name
+#  The name of the CA to manage a configuration for.
+#
+# @param anchorname
+#  The name of the CA to manage a configuration for.
+#
+# @param nowarnings
+#  Should warnings be supressed for this CA.
+#
+# @param noerrors
+#  Should errors be supressed for this CA.
+#
+# @param httptimeout
+#  The timeout for this CA.
+#
+# @param agingtolerance
+#  The delay if failures before it is considered an error.
+#
+# @param crl_url
+#  A list of URLs to download CAs from
 #
 define fetchcrl::ca(
-  $anchorname     = $title,
-  $nowarnings     = true,
-  $noerrors       = true,
-  $httptimeout    = undef,
-  $agingtolerance = undef,
-  $crl_url        = []
+  String[1] $anchorname = $title,
+  Boolean $nowarnings                            = false,
+  Boolean $noerrors                              = false,
+  Variant[Undef,Integer] $httptimeout            = undef,
+  Variant[Undef,Integer] $agingtolerance         = undef,
+  Array[Stdlib::Httpurl] $crl_url                = [],
```

Comment:
I think it was just this that didn't want to be `Optional`.  The `Variant[Undef,foo]` was just for illustration?

Reply:
I had the same internal debate with myself.
Clearly the array was wrong.
Is there any difference between
```
Variant[undef,X] $x = undef
Optional[X] $x = undef
```
I can't think of one.

Reply:
No, I believe they're identical.  But style-wise, I think we always prefer using `Optional`